### PR TITLE
Complete `python-crontab`

### PR DESCRIPTION
**Pull request by sobolevn**
Source: https://gitlab.com/doctormo/python-crontab

Source are very messy in some parts.
I've left several `TODO` items for the future, because some parts are quite complex.
I've also intentionally simplified some corner cases: for example some code might call `a = int(b)`, so I annotated `b: int | str`, because this is what docs say for multiple cases.

This was not pleasant 😒

### CODE REVIEW

File: stubs/python-crontab/crontab.pyi [hunk 1]
```diff
@@ -1,15 +1,19 @@
 import re
 import subprocess
 from _typeshed import Incomplete, Self
+from builtins import range as _range
 from collections import OrderedDict
-from collections.abc import Callable, Generator, Iterator
+from collections.abc import Callable, Iterable, Iterator
```
Comment (AlexWaygood) [inline stubs/python-crontab/crontab.pyi:6]:
```suggestion
from collections.abc import Callable, Generator, Iterable, Iterator
```

File: stubs/python-crontab/crontab.pyi [hunk 2]
```diff
@@ -48,9 +53,9 @@ class CronTab:
     @property
     def log(self) -> CronLog: ...
     @property
-    def user(self) -> str | None: ...
+    def user(self) -> _User: ...
     @property
-    def user_opt(self) -> dict[str, str]: ...
+    def user_opt(self) -> dict[str, _User]: ...
```
Comment (JelleZijlstra) [inline stubs/python-crontab/crontab.pyi:58]:
```suggestion
    def user_opt(self) -> dict[str, str]: ...
```
It doesn't return `_user` if it's a boolean or None.

File: stubs/python-crontab/crontab.pyi [hunk 3]
```diff
@@ -72,97 +80,104 @@ class CronTab:
...
+    user: _User
     valid: bool
     enabled: bool
     special: bool
-    comment: Incomplete
-    command: Incomplete
-    last_run: Incomplete
-    env: Incomplete
+    comment: str
+    command: str
```
Comment (JelleZijlstra) [inline stubs/python-crontab/crontab.pyi:107]:
```suggestion
    command: str | None
```

File: stubs/python-crontab/crontab.pyi [hunk 4]
```diff
@@ -72,97 +80,104 @@ class CronTab:
...
-    def months(self) -> CronSlice: ...
+    def months(self) -> int | str: ...
     @property
-    def dow(self) -> CronSlice: ...
+    def dow(self) -> int | str: ...
     def __len__(self) -> int: ...
-    def __getitem__(self, key: SupportsIndex) -> CronSlice: ...
-    def __lt__(self, value) -> bool: ...
-    def __gt__(self, value) -> bool: ...
+    def __getitem__(self, key: str) -> int | str: ...
```
Comment (JelleZijlstra) [inline stubs/python-crontab/crontab.pyi:162]:
```suggestion
    def __getitem__(self, key: int | str) -> int | str: ...
```

**Pull request merged**